back/src/main/java/fr/formation/itschool/masterpiece/validators/PasswordComplexityRequirement.java

```diff
@@ -0,0 +1,20 @@
+package fr.formation.itschool.masterpiece.validators;
+
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.TYPE})
+@Constraint(validatedBy = PasswordComplexityValidator.class)
+public @interface PasswordComplexityRequirement {
+  String message() default "passwordComplexityError";
```

Comment:
In fact the explicit message will be displayed by the front through the intl api

Reply:
yes, but in your log, in your stacktrace, if front fails to parse the error, this will be your default message. I would be more clearer with a sentence. 
PLUS : you can use a LocalValidatorFactoryBean to internationalize and/ or keep all your messages in one file. Hence your back end keeps the responsibility to give error message, front only display them